src/Workspaces/Core/Portable/Workspace/Solution/SolutionState.CachingFilePathComparer.cs

```diff
@@ -96,8 +96,8 @@ private unsafe int GetNonRandomizedHashCodeOrdinalIgnoreCase(string obj)
                     }
 
                     length -= 4;
-                    hash1 = (RuntimeBitOperations.RotateLeft(hash1, 5) + hash1) ^ (p0 | NormalizeToLowercase);
-                    hash2 = (RuntimeBitOperations.RotateLeft(hash2, 5) + hash2) ^ (p1 | NormalizeToLowercase);
+                    hash1 = (RotateLeft(hash1, 5) + hash1) ^ (p0 | NormalizeToLowercase);
```

Comment:
Wouldn't it be better to use `System.Numerics.BitOperations.RotateLeft`? It's a JIT intrinsic on dotnet.
We can define static extension polyfill for net472

Reply:
I  have problems with the netstandard build when I tried this:

<img width="1113" height="209" alt="image" src="https://github.com/user-attachments/assets/45937975-f6d8-40ad-9747-bcf6cf5e0fa1" />

Go to definition on it in that context yields the following from the System.Collections.Immutable package:

namespace System.Numerics
{
#if !NET
    internal static class BitOperations
    {
        [MethodImpl(MethodImplOptions.AggressiveInlining)]
        public static uint RotateLeft(uint value, int offset) => (value << offset) | (value >> (32 - offset));
    }
#endif
}

Reply:
Dur, just read your comment again and went ahead and did that in commit 2